src/textual/widgets/_digit_display.py

```diff
@@ -0,0 +1,271 @@
+from textual.app import ComposeResult
+from textual.reactive import reactive
+from textual.widget import Widget
+from textual.widgets import Static
+
+
+_character_map: dict[str, str] = {}
+
+# in the mappings below, we use underscores to make spaces more visible,
+# we will strip them out later
+_character_map[
+    "0"
+] = """
+┏━┓
+┃╱┃
+┗━┛
+"""
+
+_character_map[
+    "1"
+] = """
+·┓·
+ ┃·
+╺┻╸
+"""
+
+_character_map[
+    "2"
+] = """
+╺━┓
+┏━┛
+┗━╸
+"""
+
+_character_map[
+    "3"
+] = """
+╺━┓
+ ━┫
+╺━┛
+"""
+
+_character_map[
+    "4"
+] = """
+╻ ╻
+┗━┫
+  ╹
+"""
+
+_character_map[
+    "5"
+] = """
+┏━╸
+┗━┓
+╺━┛
+"""
+
+_character_map[
+    "6"
+] = """
+┏━╸
+┣━┓
+┗━┛
+"""
+
+_character_map[
+    "7"
+] = """
+╺━┓
+  ╹
+ ╹·
+"""
+
+_character_map[
+    "8"
+] = """
+┏━┓
+┣━┫
+┗━┛
+"""
+
+_character_map[
+    "9"
+] = """
+┏━┓
+┗━┫
+╺━┛
+"""
+
+_character_map[
+    " "
+] = """
+···
+···
+···
+"""
+
+
+_character_map[
+    "X"
+] = """
+╻ ╻
+·╋·
+╹ ╹
+"""
+
+_character_map[
+    "Y"
+] = """
+╻ ╻
+·┳·
+·╹·
+"""
+
+_character_map[
+    "Z"
+] = """
+╺━┓
+·▞·
+┗━╸
+"""
+
+
+_character_map[
+    "."
+] = """
+··
+··
+·•
+"""
+
+
+_character_map[
+    ","
+] = """
+··
+··
+·▞
+"""
+
+_character_map[
+    "+"
+] = """
+···
+╺╋╸
+···
+"""
+
+_character_map[
+    "-"
+] = """
+···
+╺━╸
+···
+"""
+
+_character_map[
+    "="
+] = """
+···
+╺━·
+╺━·
+"""
+
+_character_map[
+    "*"
+] = """
+···
+·✱·
+···
+"""
+
+
+_character_map[
+    "/"
+] = """
+··╻
+·▞·
+╹··
+"""
+
+_character_map[
+    "^"
+] = """
+·╻·
+▝·▘
+···
+"""
+
+
+_VIRTUAL_SPACE = "·"
+
+# here we strip spaces and replace virtual spaces with spaces
+_character_map = {
+    k: v.strip().replace(_VIRTUAL_SPACE, " ") for k, v in _character_map.items()
+}
+
+
+class _SingleDigitDisplay(Static):
+    digit = reactive(" ", layout=True)
+
+    DEFAULT_CSS = """
+        _SingleDigitDisplay {
+          height: 3;
+          min-width: 2;
+          max-width: 3;
+        }
+    """
+
+    def __init__(self, initial_value=" ", **kwargs):
+        super().__init__(**kwargs)
+        self.digit = initial_value
+
+    def watch_digit(self, digit: str) -> None:
```

Comment:
```suggestion
    def _watch_digit(self, digit: str) -> None:
```

Consider making this private so that it doesn't show up as part of the public API (of this private widget).